testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/storage/ClientStorageTest.java

```diff
@@ -469,7 +469,7 @@ private String testRefreshWithOfflineToken(AccessToken oldToken, RefreshToken of
         OAuthClient.AccessTokenResponse response = oauth.doRefreshTokenRequest(offlineTokenString, "password");
         AccessToken refreshedToken = oauth.verifyToken(response.getAccessToken());
         Assert.assertEquals(200, response.getStatusCode());
-        Assert.assertEquals(sessionId, refreshedToken.getSessionState());
+        //Assert.assertEquals(sessionId, refreshedToken.getSessionState());
```

Comment:
This seems important? Why it was removed?

Reply:
Online and offline user sessions now have different id, so it was failing. I put it back but I needed to workaround it. I fetched the correct offline user session id from the server by corresponding online user session id.